src/main/java/com/joss/conductor/mobile/TestListener.java

```diff
@@ -0,0 +1,45 @@
+package com.joss.conductor.mobile;
+
+import com.joss.conductor.mobile.util.ScreenShotUtil;
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+
+/**
+ * Created on 3/21/17.
+ */
+public class TestListener implements ITestListener {
+
+    public void onTestStart(ITestResult result) {
+
+    }
+
+    public void onTestSuccess(ITestResult result) {
+
+    }
+
+    public void onTestFailure(ITestResult result) {
+        Locomotive locomotive = (Locomotive) result.getInstance();
+        if (locomotive.configuration.screenshotsOnFail()) {
+                ScreenShotUtil.take(locomotive,
+                        result.getTestClass().getName() + "." + result.getMethod().getMethodName(),
+                        result.getThrowable().getMessage());
```

Comment:
The logging would have to be done separate, all this is doing is grabbing the name of the failure and naming the screenshot file to that.

The thrown exception is already being logged when it happens, if you want more logging that that (e.g. Android Device Logs, iOS, etc.) you would have to do that separately.